pinot-controller/src/main/java/com/linkedin/pinot/controller/validation/ValidationManager.java

```diff
@@ -180,6 +192,22 @@ public void runValidation() {
     LOGGER.info("Validation completed");
   }
 
+  // If we add a new broker, we want to rebuild the broker resource.
+  void rebuildBrokerResourceWhenBrokerAdded(AbstractTableConfig tableConfig, IdealState brokerIdealState) {
```

Comment:
I use this logic twice since I have to fetch the table config - I fetch the realtime table config and the offline table config with separate calls so that I can log appropriately for debugging purposes.